pulsar-common/src/main/java/com/yahoo/pulsar/common/api/Commands.java

```diff
@@ -429,11 +456,36 @@ private static ByteBuf serializeCommandSendWithSize(BaseCommand.Builder cmdBuild
             cmd.writeTo(outStream);
             cmd.recycle();
             cmdBuilder.recycle();
+            
+            //Write checksum
+            if (includeChecksum) {
+                headers.writeShort(magicCrc32c);
+                checksumReaderIndex = headers.writerIndex();
+                headers.writeZero(4); // write dummy checksum int to skip 4 bytes in write index
+            }
 
+            int metadataReaderIndex = headers.writerIndex();
             // Write metadata
             headers.writeInt(msgMetadataSize);
             msgMetadata.writeTo(outStream);
 
+            if (includeChecksum) {
+                // mark reader and write index
+                int headersReaderIndex = headers.readerIndex();
+
+                // compute checksum
+                headers.retain();
```

Comment:
Actually, as here we are creating ```DoubleByteBuf``` for checksum computation and then releasing. [DoubleByteBuf.deallocate](https://github.com/rdhabalia/pulsar/blob/dfe114e0afc45d47429637bd67939cd675c6d37f/pulsar-common/src/main/java/com/yahoo/pulsar/common/api/DoubleByteBuf.java#L363) releases internal b1 and b2 buffer twice.
```
b1.release(2);
b2.release(2);
```

So, initially: ```headers.refCnt =1``` after creating DoubleByteBuf, ```headers.Cnt=2```. But after releasing DoubleByteBuf, ```headers.Cnt=0```.